src/schema/base/literal.ts

```diff
@@ -116,17 +120,134 @@ class OptionalModifier<Schema extends BaseModifiersType<any, any>> extends BaseM
   Schema[typeof COTYPE] | undefined
 > {
   #parent: Schema
-  constructor(parent: Schema) {
+
+  /**
+   * Optional modifier validations list
+   */
+  validations: Validation<any>[]
+
+  constructor(parent: Schema, validations?: Validation<any>[]) {
     super()
     this.#parent = parent
+    this.validations = validations || []
+  }
+
+  /**
+   * Shallow clones the validations. Since, there are no API's to mutate
+   * the validation options, we can safely copy them by reference.
+   */
+  protected cloneValidations(): Validation<any>[] {
+    return this.validations.map((validation) => {
```

Comment:
Maybe it would be better to directly use [`structuredClone()`](https://developer.mozilla.org/en-US/docs/Web/API/structuredClone)?

```ts
return structuredClone(this.validations)
```

Reply:
Ohh did not know about that. Will do in a separate PR